esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -3277,4 +3326,586 @@ void WsWuHelpers::checkAndTrimWorkunit(const char* methodName, StringBuffer& inp
 
     return;
 }
+
+IFileIOStream* CWsWuFileHelper::createIOStreamWithFileName(const char* fileNameWithPath, IFOmode mode)
+{
+    Owned<IFile> wuInfoIFile = createIFile(fileNameWithPath);
+    Owned<IFileIO> wuInfoIO = wuInfoIFile->open(mode);
+    if (!wuInfoIO)
+        throw MakeStringException(ECLWATCH_CANNOT_OPEN_FILE, "Failed to open %s.", fileNameWithPath);
+    return createIOStream(wuInfoIO);
+}
+
+void CWsWuFileHelper::writeToFile(const char* fileName, size32_t contentLength, const void* content)
+{
+    if (isEmptyString(fileName))
+        throw MakeStringException(ECLWATCH_CANNOT_COMPRESS_DATA, "File name not specified.");
+    Owned<IFile> wuInfoIFile = createIFile(fileName);
+    Owned<IFileIO> wuInfoIO = wuInfoIFile->open(IFOcreate);
+    if (wuInfoIO)
+        wuInfoIO->write(0, contentLength, content);
+}
+
+void CWsWuFileHelper::writeToFileIOStream(const char* folder, const char* file, MemoryBuffer& mb)
+{
+    VStringBuffer fileNameWithPath("%s%c%s", folder, PATHSEPCHAR, file);
+    CWsWuFileHelper helper(nullptr);
+    Owned<IFileIOStream> outIOS = helper.createIOStreamWithFileName(fileNameWithPath.str(), IFOcreate);
+    if (outIOS)
+        outIOS->write(mb.length(), mb.toByteArray());
```

Comment:
Right now, the answer is no because the createIOStreamWithFileName() will throw an exception if failed before new. But, I am going to add a log statement here in case the  createIOStreamWithFileName() is changed in future.